pkgs/development/libraries/science/math/openblas/default.nix

```diff
@@ -208,15 +208,21 @@ EOF
     done
 
     # Setup symlinks for blas / lapack
```

Comment:
I think you can avoid the rebuild by adding a newline after this comment in the string. I think it should be possible to make this change without changing the string content here for `isLinux && enableShared`.

Reply:
Unfortunately, it seems like the writing the pkgconfig file above messes this up.

On `master`, this is what the `postInstall` looks like:

```console
$ nix repl ./.
nix-repl> builtins.trace openblas.postInstall 3
trace:     # Write pkgconfig aliases. Upstream report:
    # https://github.com/xianyi/OpenBLAS/issues/1740
    for alias in blas cblas lapack; do
      cat <<EOF > $out/lib/pkgconfig/$alias.pc
Name: $alias
Version: 0.3.17
Description: $alias provided by the OpenBLAS package.
Cflags: -I$out/include
Libs: -L$out/lib -lopenblas
EOF
    done

    # Setup symlinks for blas / lapack
    ln -s $out/lib/libopenblas.so $out/lib/libblas.so
    ln -s $out/lib/libopenblas.so $out/lib/libcblas.so
    ln -s $out/lib/libopenblas.so $out/lib/liblapack.so
    ln -s $out/lib/libopenblas.so $out/lib/liblapacke.so
ln -s $out/lib/libopenblas.so $out/lib/libblas.so.3
ln -s $out/lib/libopenblas.so $out/lib/libcblas.so.3
ln -s $out/lib/libopenblas.so $out/lib/liblapack.so.3
ln -s $out/lib/libopenblas.so $out/lib/liblapacke.so.3

3
```

Here's what it looks like with this PR:

```console
$ nix repl ./.
nix-repl> builtins.trace openblas.postInstall 3
trace:     # Write pkgconfig aliases. Upstream report:
    # https://github.com/xianyi/OpenBLAS/issues/1740
    for alias in blas cblas lapack; do
      cat <<EOF > $out/lib/pkgconfig/$alias.pc
Name: $alias
Version: 0.3.17
Description: $alias provided by the OpenBLAS package.
Cflags: -I$out/include
Libs: -L$out/lib -lopenblas
EOF
    done

    # Setup symlinks for blas / lapack
ln -s $out/lib/libopenblas.so $out/lib/libblas.so
ln -s $out/lib/libopenblas.so $out/lib/libcblas.so
ln -s $out/lib/libopenblas.so $out/lib/liblapack.so
ln -s $out/lib/libopenblas.so $out/lib/liblapacke.so
ln -s $out/lib/libopenblas.so $out/lib/libblas.so.3
ln -s $out/lib/libopenblas.so $out/lib/libcblas.so.3
ln -s $out/lib/libopenblas.so $out/lib/liblapack.so.3
ln -s $out/lib/libopenblas.so $out/lib/liblapacke.so.3

3
```

It makes sense that with this PR, the first set of 4 `ln` lines lines are no longer indented, because they are now in their own multi-line string block.  They no longer get their indentation set by the content of the pkgconfig file.

I'm sure there would be _some_ way of writing this Nix expression so that everything is indented correctly and it doesn't trigger any rebuilds, but I'm not sure it would be worth it.

Reply:
Ah right, that would be really finnicky to do with zero rebuilds.